Dotnetcore

TrueUO opened the pull request:
DO NOT MERGE

- We are keeping this PR open until it is confirmed stable on a few shards. We really appreciate any feedback or PRs from the community on this one.

Argalep committed:
dotnetcore master

Argalep committed:
fix

Argalep committed:
fix

Argalep committed:
* restart problem fix

Argalep committed:
* add sh build

Argalep committed:
cmd or operator fix

Voxpire committed:
Port from delegate BeginInvoke/EndInvoke to Tasks

Argalep committed:
fix

Argalep committed:
fix

Argalep committed:
Rename Publish.cmd to publish.cmd

Argalep committed:
Rename Publish.sh to publish.sh

Voxpire committed:
Compile fix for ArchivedSaves

Argalep committed:
Merge branch 'master' into dotnetcore

Argalep committed:
32bit remove

Argalep committed:
fix

deccer commented:
Edit: Turned my comment into a review.

TrueUO committed:
Merge branch 'master' into dotnetcore

deccer commented on Application/ServUO.csproj:
```diff
@@ -1,18 +1,18 @@
     <StartupObject>ServUO.ServUO</StartupObject>
     <RootNamespace>Server</RootNamespace>
+	<PublishDir>..\..\Distribution</PublishDir>
+	<OutDir>..\..\Distribution</OutDir>
```
Those should be `..\Distribution` instead

deccer commented on Scripts/Scripts.csproj:
```diff
@@ -2,14 +2,16 @@
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <RootNamespace>Server</RootNamespace>
+	<PublishDir>..\..\Distribution</PublishDir>
+	<OutDir>..\..\Distribution</OutDir>
```
Those should be `..\Distribution` instead

deccer commented on Server/Server.csproj:
```diff
@@ -1,15 +1,16 @@
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <RootNamespace>Server</RootNamespace>
+	<PublishDir>..\..\Distribution</PublishDir>
+	<OutDir>..\..\Distribution</OutDir>
```
Those should be `..\Distribution` instead

deccer commented on Ultima/Ultima.csproj:
```diff
@@ -2,14 +2,16 @@
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <RootNamespace>Server</RootNamespace>
+	<PublishDir>..\..\Distribution</PublishDir>
+	<OutDir>..\..\Distribution</OutDir>
```
Those should be `..\Distribution` instead

TrueUO replied:
Thanks a lot for the help @deccer!

deccer reviewed (changes requested):
I decided to do it properly, and turn the comment into a review. I hope you don't mind.

TrueUO committed:
Merge branch 'master' into dotnetcore

Argalep committed:
linux restart fix

Argalep committed:
Merge branch 'dotnetcore' of https://github.com/TrueUO/TrueUO into dotnetcore

Argalep committed:
csproj path fix.

Argalep committed:
yml core

TrueUO committed:
Merge branch 'master' into dotnetcore

Argalep committed:
email fix

Argalep committed:
Merge branch 'dotnetcore' of https://github.com/TrueUO/TrueUO into dotnetcore

Argalep committed:
fix email

Argalep committed:
crashguard restart fix.

DocKnight86 committed:
TravisAttempt1

TrueUO committed:
Update .travis.yml

TrueUO committed:
Update .travis.yml

TrueUO committed:
Update .travis.yml

TrueUO committed:
Update .travis.yml

TrueUO committed:
Update .travis.yml

TrueUO committed:
Update .travis.yml

DocKnight86 committed:
Removed unneeded using
Changed Framework to Core for crash reports.

DocKnight86 committed:
Currently the changes made to ArchivedSaves added a 2x length to normal save times, even if ArchivedSaves was set to false.

Reverted removes the issue but at the moment will break ArchivedSaves. (not the autosave or backups)

kevin-10 commented:
When archived saves are set to false, none of that code is even called. Are we absolutely sure that's what was causing the extra save time?

TrueUO commented:
> When archived saves are set to false, none of that code is even called. Are we absolutely sure that's what was causing the extra save time?

Even archivedsaves are off, it does still seem to check if it is off, during each save causing the new delay when trying to make it compatible.

When I reverted to the old method it worked again, no 2x save issues. It has to do with when we tried to update it recently.

TrueUO commented:
@Argalep has a potential fix just waiting for him to push it.

kevin-10 committed:
Merge branch 'master' into dotnetcore

DocKnight86 committed:
Fixes

Re-applied Vorspire original changes made to ArchivedSaves along with Argaleps !Enabled @ WaitForTaskCompletion

TrueUO commented:
I pushed @Argalep suggestion and re-added @Vorspire original changes to ArchivedSaves. 

Have not yet tested on Heritage.

DocKnight86 committed:
Cleanup

- Simplified everything that could be.
- Removed un-needed code.

DocKnight86 committed:
Ultima cleanup

DocKnight86 committed:
More cleanup

DocKnight86 committed:
Remove unneeded using.System

TrueUO committed:
Merge branch 'master' into dotnetcore

TrueUO commented:
Just closing this for now due to holding up checks on the main. Is it a sonar issue?
